### PR TITLE
[kernel] API to get the number of module parameters

### PR DESCRIPTION
**Pull request by gares**
We provide a few APIs to access the module structure being declared interactively.
This in turn can be used by clients of the libobject API to support modules but not functors, for example.

### CODE REVIEW

File: kernel/safe_typing.mli [hunk 1]
```diff
@@ -169,6 +169,9 @@ val add_module_parameter :
   MBId.t -> Entries.module_struct_entry -> Declarations.inline ->
     Mod_subst.delta_resolver safe_transformer
 
+(** returns the number of module parameters *)
+val module_num_parameters : safe_environment -> int
```
Comment (ppedrot) [inline kernel/safe_typing.mli:173]: Don't you want to also return whether we are in a module type or a module?

Comment (gares) [reply kernel/safe_typing.mli:173]: done

**Review by ppedrot — APPROVED** [review]

File: kernel/safe_typing.mli [hunk 2]
```diff
@@ -169,6 +169,12 @@ val add_module_parameter :
   MBId.t -> Entries.module_struct_entry -> Declarations.inline ->
     Mod_subst.delta_resolver safe_transformer
 
+(** returns the number of module parameters *)
```
Comment (SkySkimmer) [inline kernel/safe_typing.mli:172]: Should state whether it's the parameters of the topmost module or all of them. eg what is returned after
~~~coq
Module F(A:T).
  Module Inner.
~~~
?
What about
~~~coq
Module F(A:T).
  Module F'(B:TT).
~~~
?

File: kernel/safe_typing.mli [hunk 3]
```diff
@@ -169,6 +169,12 @@ val add_module_parameter :
...
+val module_num_parameters : safe_environment -> int
+
+(** returns true if the current module is a module type *)
+val module_is_modtype : safe_environment -> bool
```
Comment (SkySkimmer) [inline kernel/safe_typing.mli:176]: Should Lib.is_modtype be implemented in terms of this? Kinda weird to have both.

File: kernel/safe_typing.mli (hunk 2 above)
Comment (gares) [reply kernel/safe_typing.mli:172]: Thkns, I need  to get the full list to know if I'm a functor.

**Comment by silene** [comment]
If I understand correctly, this pull request does not solve the issue, so I am marking it as draft and postponing it to 8.15.

**Comment by gares** [comment]
Ok

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**